test/nightly/deploy.test.ts

```diff
@@ -120,41 +120,50 @@ suite('Create Web App and deploy', function (this: Mocha.Suite): void {
         });
     }
 
-    async function testCreateWebAppAndDeploy(os: string, promptForOs: boolean, runtime: string, workspacePath: string, expectedVersion: string, zipFile?: string): Promise<void> {
+    async function testCreateWebAppAndDeploy(os: string, promptForOs: boolean, runtime: string, _workspacePath: string, _expectedVersion: string, _zipFile?: string): Promise<void> {
         const resourceName: string = getRandomHexString();
-        const resourceGroupName = getRandomHexString();
-        resourceGroupsToDelete.push(resourceGroupName);
+        const resourceGroupName: string = azcodeResourcePrefix + getRandomHexString(6);
+        resourceGroupsToDelete.add(resourceGroupName);
 
-        const testInputs: (string | RegExp)[] = [resourceName, '$(plus) Create new resource group', resourceGroupName, runtime];
+        const testInputs: (string | RegExp)[] = [/Manual\sTest/i, resourceName, '$(plus) Create new resource group', resourceGroupName, runtime];
```

Comment:
With the service principal, there should only be one subscription that can be selected which results in an auto-selection. I think that this isn't being checked right now since you have the deploy tests commented out, but this probably won't work.

Reply:
> With the service principal, there should only be one subscription that can be selected which results in an auto-selection.

This is what I assumed as well, however, it seems that the auto-selection wasn't working even when there was only one subscription in the filter.  A better long term solution probably needs to be to investigate why the subscription isn't being automatically selected.

> I think that this isn't being checked right now since you have the deploy tests commented out, but this probably won't work.

I think it should be working-- I checked remotely and everything was passing.  And one of the test assertions is this:
```typescript
const createdApp: Site | undefined = await tryGetWebApp(webSiteClient, resourceGroupName, resourceName);
assert.ok(createdApp);
```

Reply:
Reinvestigated this and think we will need to talk more about this offline, see this [comment](https://github.com/microsoft/vscode-azureappservice/pull/2653/files#r1696108931).